### DISCUSSION TITLE
Unable to autocomplete the "asp" command

### DISCUSSION THREAD

**Discussion by TaiSHiNet** [Aloxaf/fzf-tab#418] [Q&A] [answered]
I'm unsure if this is the right place to ask, but I've figured I might as well try:
I'm using the following oh-my-zsh plugin https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/aws

And regular completion works. But when I add the fzf plugin I get 2 behaviors:
* calling asp **<tab> it tries to list all files
* calling asp <tab> uses default zsh selection mechanism

Added `_fzf_complete_asp` and it works, but it doesn't auto complete/tab the unambiguous prefix (which does work with this plugin with tools like `awscli`)

Any guidance is greatly appreciated!

**Comment by Aloxaf** [comment, marked as answer]
The `aws` plugin utilizes zsh's outdated `compctl` command, which is not supported by fzf-tab.

https://github.com/ohmyzsh/ohmyzsh/blob/c891b7e6ada5427f4c8cca05c8ed1ae33cb49943/plugins/aws/aws.plugin.zsh#L255

However, considering that complex `compctl` parameters are not used by the code above, you may want to try the following script:

```zsh
_aws_asp() {
  local -a reply
  _aws_profiles
  compadd -a reply
}

compdef _aws_asp asp acp aws_change_access_key
```

> Added _fzf_complete_asp and it works, but it doesn't auto complete/tab the unambiguous prefix (which does work with this plugin with tools like awscli)

Sorry, but what's `_fzf_complete_asp`?

**Reply by TaiSHiNet** [reply]
Your solution works like a charm! I bashed (or should I say zsh'd?) my head against this for a while.
I might need to learn zsh a bit more in depth to figure out what your solution does, and maybe add an improvement to the aws plugin.

`_fzf_complete_COMMAND` is used for [custom command completion](https://github.com/junegunn/fzf?tab=readme-ov-file#custom-fuzzy-completion) (I was trying to make it work any way I could). 

Thanks for taking the time to help out!
